pkg/ebpf/events_pipeline.go

```diff
@@ -218,52 +218,57 @@ func (t *Tracee) decodeEvents(outerCtx context.Context, sourceChan chan []byte)
 				}
 			}
 
-			evt := trace.Event{
-				Timestamp:             int(ctx.Ts),
-				ThreadStartTime:       int(ctx.StartTime),
-				ProcessorID:           int(ctx.ProcessorId),
-				ProcessID:             int(ctx.Pid),
-				ThreadID:              int(ctx.Tid),
-				ParentProcessID:       int(ctx.Ppid),
-				HostProcessID:         int(ctx.HostPid),
-				HostThreadID:          int(ctx.HostTid),
-				HostParentProcessID:   int(ctx.HostPpid),
-				UserID:                int(ctx.Uid),
-				MountNS:               int(ctx.MntID),
-				PIDNS:                 int(ctx.PidID),
-				ProcessName:           string(bytes.TrimRight(ctx.Comm[:], "\x00")),
-				HostName:              string(bytes.TrimRight(ctx.UtsName[:], "\x00")),
-				CgroupID:              uint(ctx.CgroupID),
-				ContainerID:           containerData.ID,
-				Container:             containerData,
-				Kubernetes:            kubernetesData,
-				EventID:               int(ctx.EventID),
-				EventName:             eventDefinition.Name,
-				MatchedPoliciesKernel: ctx.MatchedPolicies,
-				ArgsNum:               int(argnum),
-				ReturnValue:           int(ctx.Retval),
-				Args:                  args,
-				StackAddresses:        stackAddresses,
-				ContextFlags:          flags,
-				Syscall:               syscall,
-			}
+			// get an event pointer from the pool
+			evt := t.eventsPool.Get().(*trace.Event)
+			// populate all the fields of the event used in this stage, and reset the rest
+			evt.Timestamp = int(ctx.Ts)
+			evt.ThreadStartTime = int(ctx.StartTime)
+			evt.ProcessorID = int(ctx.ProcessorId)
+			evt.ProcessID = int(ctx.Pid)
+			evt.ThreadID = int(ctx.Tid)
+			evt.ParentProcessID = int(ctx.Ppid)
+			evt.HostProcessID = int(ctx.HostPid)
+			evt.HostThreadID = int(ctx.HostTid)
+			evt.HostParentProcessID = int(ctx.HostPpid)
+			evt.UserID = int(ctx.Uid)
+			evt.MountNS = int(ctx.MntID)
+			evt.PIDNS = int(ctx.PidID)
+			evt.ProcessName = string(bytes.TrimRight(ctx.Comm[:], "\x00"))
+			evt.HostName = string(bytes.TrimRight(ctx.UtsName[:], "\x00"))
+			evt.CgroupID = uint(ctx.CgroupID)
+			evt.ContainerID = containerData.ID
+			evt.Container = containerData
+			evt.Kubernetes = kubernetesData
+			evt.EventID = int(ctx.EventID)
+			evt.EventName = eventDefinition.Name
+			evt.MatchedPoliciesKernel = ctx.MatchedPolicies
+			evt.MatchedPoliciesUser = 0
+			evt.MatchedPolicies = []string{}
+			evt.ArgsNum = int(argnum)
+			evt.ReturnValue = int(ctx.Retval)
+			evt.Args = args
+			evt.StackAddresses = stackAddresses
+			evt.ContextFlags = flags
+			evt.Syscall = syscall
+			evt.Metadata = nil
```

Comment:
Why `nil` and not `&trace.Metadata{}` as before?

Reply:
To be discussed, but what I think is that the current instantiation `evt := trace.Event{`, sets Metadata to nil. `&trace.Metadata{}` doesn't initialise its map Properties field `map[string]interface{}`. If we go that way to avoid nil pointer deref and so, that can be also instantiated prematurely and unnecessarily if we instantiate Metadata fully. WDYT?

Reply:
Actually it may be better to leave it this way, since we define it as `omitempty` in the `json` tag of trace.Event.